msal/application.py

```diff
@@ -633,16 +633,18 @@ def _acquire_token_silent_by_finding_rt_belongs_to_me_or_my_family(
                 **kwargs)
             if at and "error" not in at:
                 return at
+        last_resp = None
         if app_metadata.get("family_id"):  # Meaning this app belongs to this family
-            at = self._acquire_token_silent_by_finding_specific_refresh_token(
+            last_resp = self._acquire_token_silent_by_finding_specific_refresh_token(
                 authority, scopes, dict(query, family_id=app_metadata["family_id"]),
                 **kwargs)
-            if at and "error" not in at:
-                return at
+            if last_resp and "error" not in last_resp:
+                return last_resp
         # Either this app is an orphan, so we will naturally use its own RT;
         # or all attempts above have failed, so we fall back to non-foci behavior.
         return self._acquire_token_silent_by_finding_specific_refresh_token(
-            authority, scopes, dict(query, client_id=self.client_id), **kwargs)
+            authority, scopes, dict(query, client_id=self.client_id),
+            **kwargs) or last_resp
```

Comment:
Yes, this is also how I would implement it. One question though. Have we tested how the new test case is (supposed to) failing if this `or last_resp` part is not present? I briefly tried that just now, but it seems still passing. Did I miss anything?

Reply:
I just tested it without `or last_resp` and it fails for me, not sure what are you missing ?

Reply:
My bad. (The things missing is a better pair of glasses, perhaps.)